### PR TITLE
Refactor create_or_update_pull_request.rb

### PR DESCRIPTION
**Pull request by dzivalli**
**Main change: refactored `create_or_update_pull_request.rb` without changing current functionality.** 
- refactored `perform` method because it's hard to read and figure out what's going on there, it has a lot of responsibilities and does many different things
- moved some common variables into instance variables

**Misc:**
- ignore .idea
- fix migrations
- rubocop warnings

### CODE REVIEW

File: spec/mediators/create_or_update_pull_request_spec.rb [hunk 1]
```diff
@@ -89,13 +89,13 @@
       let!(:pull_request) { FactoryBot.create :pull_request, number: 9876, repository: repo }
       let!(:gen_reviewers) { FactoryBot.create_list :reviewer, 2, pull_request: pull_request }
 
-      let(:r1_name) { SecureRandom.hex }
+      let(:r1_name) { "r1_name" }
```
Comment (dzivalli) [inline spec/mediators/create_or_update_pull_request_spec.rb:92]: hard to debug with random strings..

File: app/mediators/create_or_update_pull_request.rb [hunk 2]
```diff
@@ -3,22 +3,56 @@
 class CreateOrUpdatePullRequest
   include GithubApi
 
+  attr_reader :pull_request,
+    :body,
```
Comment (dzivalli) [inline app/mediators/create_or_update_pull_request.rb:7]: lines 6-7 look strange but it's current linter setup

File: db/migrate/20151217010127_create_pull_requests.rb [hunk 3]
```diff
@@ -1,4 +1,4 @@
-class CreatePullRequests < ActiveRecord::Migration
+class CreatePullRequests < ActiveRecord::Migration[5.0]
```
Comment (dzivalli) [inline db/migrate/20151217010127_create_pull_requests.rb:1]: first migration with version had 5.0

**Comment by aergonaut** [comment]
@dzivalli Thank you for your PR! However, at this time, I am not interested in accepting refactoring PRs. PRs implementing new features or fixing bugs are welcomes, but I am not interested PRs that are only reorganizing existing code.

**Pull request closed without merging**